File: sorting/valid-anagram.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length() !=t.length()){
            return false;
        }

        HashMap<Character, Integer> map1 = new HashMap<>();
        HashMap<Character, Integer> map2 = new HashMap<>();

        // (character, frequency)

        for (int i = 0;i<s.length();i++){

            char char_s = s.charAt(i);
            char char_t = t.charAt(i);

            map1.put(char_s, map1.getOrDefault(char_s, 0)+1);
            map2.put(char_t, map2.getOrDefault(char_t, 0)+1);

        }


        return map1.equals(map2);
    }
}

File: string/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        dict1 = {}
        dict2 = {}

        if len(s)!= len(t):
            return False

        for i in s:
            if i in dict1:
                dict1[i]+=1
            else:
                dict1[i]=1
        
        for i in t:
            if i in dict2:
                dict2[i]+=1
            else:
                dict2[i]=1

        if dict1 == dict2:
            return True
        return False
